src/time/span.cr

```diff
@@ -217,24 +263,38 @@ struct Time::Span
 
   def *(number : Number)
     # TODO check overflow
-    Span.new(ticks * number)
+    Span.new(
+      seconds: total_seconds_i * number,
+      nanoseconds: nanoseconds * number,
```

Comment:
Maybe convert `nanoseconds` to an Int64 here? It's fairly easy to overflow the maximum 999_999_999 with number a mere 2.15, whereas with an Int64 `number` can be as big as 9223372046.08.